java-checks/src/main/java/org/sonar/java/checks/UnconditionalJumpStatementCheck.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.java.cfg.CFG;
+import org.sonar.java.cfg.CFG.Block;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.BlockTree;
+import org.sonar.plugins.java.api.tree.BreakStatementTree;
+import org.sonar.plugins.java.api.tree.ContinueStatementTree;
+import org.sonar.plugins.java.api.tree.ForEachStatement;
+import org.sonar.plugins.java.api.tree.ForStatementTree;
+import org.sonar.plugins.java.api.tree.ListTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.ReturnStatementTree;
+import org.sonar.plugins.java.api.tree.StatementTree;
+import org.sonar.plugins.java.api.tree.SyntaxToken;
+import org.sonar.plugins.java.api.tree.ThrowStatementTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+@Rule(key = "S1751")
+public class UnconditionalJumpStatementCheck extends IssuableSubscriptionVisitor {
+
+  private static final Tree.Kind[] LOOP_KINDS = {
+    Tree.Kind.DO_STATEMENT,
+    Tree.Kind.WHILE_STATEMENT,
+    Tree.Kind.FOR_STATEMENT,
+    Tree.Kind.FOR_EACH_STATEMENT
+  };
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(
+      Tree.Kind.BREAK_STATEMENT,
+      Tree.Kind.CONTINUE_STATEMENT,
+      Tree.Kind.RETURN_STATEMENT,
+      Tree.Kind.THROW_STATEMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    Tree parent = tree.parent();
+    while (parent.is(Tree.Kind.BLOCK)) {
+      parent = parent.parent();
+    }
+
+    if (!parent.is(LOOP_KINDS)) {
+      return;
+    }
+
+    if (tree.is(Tree.Kind.CONTINUE_STATEMENT) || !canExecuteMoreThanOnce(parent)) {
+      SyntaxToken jumpKeyword = jumpKeyword(tree);
+      reportIssue(jumpKeyword, String.format("Remove this \"%s\" statement or make it conditional.", jumpKeyword.text()));
+    }
+  }
+
+  private static boolean canExecuteMoreThanOnce(Tree loopTree) {
```

Comment:
I would name that method executeUnconditionnally to make things clearer on reporting side. and add a comment inside the method to explain that executing unconditionally means that we cannot find a path in the CFG that goes twice through this instruction.